scripts/run_backend_tests_test.py

```diff
@@ -0,0 +1,888 @@
+
+        swap_subprocess_run = self.swap(subprocess, 'run', mock_subprocess_run)
+        with swap_subprocess_run:
+            returned_output, coverage = run_backend_tests._check_coverage( # type: ignore[no-untyped-call] # pylint: disable=protected-access
+                True)
+
+        self.assertEqual(returned_output, coverage_report_output)
+        self.assertEqual(coverage, 100)
+
+    def test_failure_to_run_test_tasks_throws_error(self) -> None:
+        with self.swap_install_third_party_libs:
+            from scripts import run_backend_tests
+
+        def mock_run_shell_cmd(*unused_args: Any, **unused_kwargs: Any) -> None:
+            raise Exception('XYZ error.')
+        swap_run_shell_cmd = self.swap(
+            run_backend_tests, 'run_shell_cmd', mock_run_shell_cmd)
+        swap_hostname = self.swap(socket, 'gethostname', lambda: 'IamEzio')
+        swap_getpid = self.swap(os, 'getpid', lambda: 12345)
+
+        task = run_backend_tests.TestingTaskSpec( # type: ignore[no-untyped-call]
+            'scripts.run_backend_tests_test', False)
+        with swap_run_shell_cmd, swap_hostname, swap_getpid:
+            with self.assertRaisesRegex(Exception, 'XYZ error.'):
+                task.run() # type: ignore[no-untyped-call]
+
+    def test_tasks_run_again_if_race_condition_occurs(self) -> None:
+        with self.swap_install_third_party_libs:
+            from scripts import run_backend_tests
+
+        def mock_run_shell_cmd(*unused_args: Any, **unused_kwargs: Any) -> str:
+            if self.call_count == 1:
+                return 'Task result'
+            self.call_count = 1
+            raise Exception('ev_epollex_linux.cc')
+        swap_run_shell_cmd = self.swap(
+            run_backend_tests, 'run_shell_cmd', mock_run_shell_cmd)
+        swap_hostname = self.swap(socket, 'gethostname', lambda: 'IamEzio')
+        swap_getpid = self.swap(os, 'getpid', lambda: 12345)
+        swap_check_coverage = self.swap(
+            run_backend_tests, '_check_coverage',
+            lambda *unused_args, **unused_kwargs: ('Coverage report', 100.00))
+
+        task = run_backend_tests.TestingTaskSpec( # type: ignore[no-untyped-call]
+            'scripts.run_backend_tests_test', True)
+        with swap_run_shell_cmd, swap_hostname, swap_getpid:
+            with swap_check_coverage:
+                results = task.run() # type: ignore[no-untyped-call]
+
+        self.assertIn('Task result', results[0].messages)
+        self.assertIn('Coverage report', results[0].messages)
+
+    def test_invalid_file_in_task_returns_empty_report(self) -> None:
+        with self.swap_install_third_party_libs:
+            from scripts import run_backend_tests
+
+        def mock_run_shell_cmd(*unused_args: Any, **unused_kwargs: Any) -> str:
+            if self.call_count == 1:
+                return 'Task result'
+            self.call_count = 1
+            raise Exception('ev_epollex_linux.cc')
```

Comment:
Should this be a class function that gets re-used since there seem to be several copies of this code?

Reply:
It is used in only 4 tests and is used to check different scenarios while executing the tasks. I think we can let it be a normal function rather than making it a class function in `setup` method. WDYT?